meta-ostro/classes/ostro-image-extensions.bbclass

```diff
@@ -0,0 +1,21 @@
+# Class that provides additional tools and methods for image conversions.
+
+COMPRESSIONTYPES_append = " vdi bmap md5sum sha1sum sha224sum sha256sum sha384sum sha512sum"
+COMPRESS_CMD_vdi = "qemu-img convert -O vdi ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.vdi"
+COMPRESS_CMD_bmap = "bmaptool create ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} -o ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.bmap"
+COMPRESS_CMD_md5sum = "md5sum ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} > ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.md5sum"
+COMPRESS_CMD_sha1sum = "sha1sum ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} > ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.sha1sum"
+COMPRESS_CMD_sha224sum = "sha224sum ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} > ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.sha224sum"
+COMPRESS_CMD_sha256sum = "sha256sum ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} > ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.sha256sum"
+COMPRESS_CMD_sha384sum = "sha384sum ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} > ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.sha384sum"
+COMPRESS_CMD_sha512sum = "sha512sum ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} > ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.sha512sum"
+
+COMPRESS_DEPENDS_vdi = "qemu-native"
+COMPRESS_DEPENDS_bmap = "bmap-tools-native"
+COMPRESS_DEPENDS_md5sum = "coreutils-native"
```

Comment:
Ideally, this should be converted to use python implementation of hashlib functions, but I'm not sure how to specify that in bbclass syntax. if you have idea how to do that, please share.

As for coreutils-native, we definetely want to get @rpurdie, @rossburton feedback, but I'd prefer to play safe. if somebody wants to skip build of it, it's possible to set "assume provided" in local.conf for it.